Remove unused hydrostatic code.

middlecoff opened the pull request:
## Proposed changes

Describe the big picture of your changes here to communicate what they accomplish and why this pull request should be accepted.
Removed unused hydrostatic code to simplify the remaining code. The unused code was in IF statements controlled by the logical parameter "hydrostatic" which was set to false.

If it fixes a bug or resolves a feature request, be sure to link to that issue.

If it depends on other pull requests, be sure to list and link those dependencies.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in all the boxes that apply_

- [ ] Bugfix (Change which fixes an issue)
- [ ] New feature (Change which adds functionality)
- [x] Enhancement to existing functionality (non-breaking change which improves existing code or documentation)
- [ ] Breaking change (fix or feature that changes results and requires updates to test baselines)
- [ ] Documentation Update

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/NOAA-GSL/SENA-c_sw/blob/develop/CONTRIBUTING.md) document
- [x] I have ensured all my changes contribute to a common purpose
- [x] I have verified my change does not add debug code
- [x] I have verified my change does not add code that is commented out
- [x] I have verified unit tests pass locally with my changes
- [x] I have verified my changes adhere to style guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

middlecoff committed:
Remove unused hydrostatic code.

christopherwharrop-noaa commented on ref/src/sw_core.F90:
```diff
@@ -172,34 +171,6 @@ subroutine c_sw(sw_corner, se_corner, nw_corner, ne_corner,       &
-      enddo
-#endif
-    else
       if (grid_type < 3) then
```
Please also adjust the indentation of the `else` part of the branch that is now no longer inside an `if`.

christopherwharrop-noaa commented on ref/src/sw_core.F90:
```diff
@@ -219,39 +190,12 @@ subroutine c_sw(sw_corner, se_corner, nw_corner, ne_corner,       &
-      enddo
-    else
+
       if (grid_type < 3) then
```
Same indentation request here.  Please readjust the indentation of the former `else` branch to match the new structure of the code.

middlecoff replied:
Fixed indentation.

christopherwharrop-noaa reviewed (changes requested):
Thank you for this work.  Simple to follow and good to clean this up.  My only request is that the indentation of the `else` part that remained after you removed the `if` be properly indented to match the new structure of the code.

christopherwharrop-noaa committed:
Merge branch 'develop' into feature/remove-hydrostatic-code

middlecoff commented:
Hi Chris,

I thought of fixing the indentation but decided not to because fixing the
indentation shows up as a lot more changes - I guess more changes is not an
issue?

So do I just make the indentation changes in my existing
feature/remove-hydrostatic-code branch and then do a git push again?

Jacques


On Wed, Jan 13, 2021 at 3:48 PM Christopher Harrop <notifications@github.com>
wrote:

> *@christopherwharrop-noaa* requested changes on this pull request.
>
> Thank you for this work. Simple to follow and good to clean this up. My
> only request is that the indentation of the else part that remained after
> you removed the if be properly indented to match the new structure of the
> code.
> ------------------------------
>
> In ref/src/sw_core.F90
> <https://github.com/NOAA-GSL/SENA-c_sw/pull/25#discussion_r556925779>:
>
> > -#else
> -      do j = js-1, jep1
> -        do i = is-1, ie+2
> -          if ( ut(i, j) > 0. ) then
> -            fx1(i, j) = delp(i-1, j)
> -            fx(i, j) = pt(i-1, j)
> -          else
> -            fx1(i, j) = delp(i, j)
> -            fx(i, j) = pt(i, j)
> -          endif
> -          fx1(i, j) = ut(i, j) * fx1(i, j)
> -          fx(i, j) = fx1(i, j) * fx(i, j)
> -        enddo
> -      enddo
> -#endif
> -    else
>        if (grid_type < 3) then
>
> Please also adjust the indentation of the else part of the branch that is
> now no longer inside an if.
> ------------------------------
>
> In ref/src/sw_core.F90
> <https://github.com/NOAA-GSL/SENA-c_sw/pull/25#discussion_r556927082>:
>
> > -        enddo
> -      enddo
> -      do j = js-1, jep1
> -        do i = is-1, iep1
> -          delpc(i, j) = delp(i, j) + (fx1(i, j) - fx1(i+1, j) + fy1(i, j) - &
> -                        fy1(i, j+1)) * rarea(i,j)
> -#ifdef SW_DYNAMICS
> -          ptc(i, j) = pt(i, j)
> -#else
> -          ptc(i, j) = (pt(i, j) * delp(i, j) + (fx(i, j) - fx(i+1, j) + &
> -                       fy(i, j) - fy(i, j+1)) * rarea(i, j)) / delpc(i, j)
> -#endif
> -        enddo
> -      enddo
> -    else
> +
>        if (grid_type < 3) then
>
> Same indentation request here. Please readjust the indentation of the
> former else branch to match the new structure of the code.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-GSL/SENA-c_sw/pull/25#pullrequestreview-567719557>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFMA7HJ6553FQIA72ZGHAGDSZYPK5ANCNFSM4WBQOWCQ>
> .
>

christopherwharrop-noaa commented:
@middlecoff  - Your thinking was for sure in the right frame of mind.  But we do want the indentation to be correct.  Just commit the changes locally and do another push and it will automatically update this pull request.

christopherwharrop-noaa commented:
@middlecoff - As a side note, I recommend not using email to discuss pull request changes.  You will get more out of this process by using the comment/discussion tools on GitHub directly.  For example, when discussing code, there are many instances where formatting is key to communication and which cannot be accomplished via response by email.  Also, GitHub provides the @ syntax for addressing/notifying specific people, as I have done in this message.

middlecoff commented:
Since this is a follow on request it is different from last time - have I submitted a second pull request?

christopherwharrop-noaa commented:
@middlecoff - No.  It is the same pull request, but I'm not seeing your push.  Did you make the requested change and push again?

robgpita reviewed (commented):
@middlecoff Looks good! I don't have anything more to add to Chris' requests. The more code you remove, the less I have to translate!

robgpita commented:
@middlecoff If you made changes locally, you have to `git add`, `git commit` & `git push`, which should update this PR.

LyndStringer reviewed (dismissed):
Looks good.  I wrote an issue with the ifdef(s).

christopherwharrop-noaa commented:
@middlecoff - I will approve and merge as soon as the indentation is addressed.  As @robgonzalezpita noted, you need to make the changes locally, then push again to your branch.  That will automatically update this pull request.

christopherwharrop-noaa committed:
Merge branch 'develop' into feature/remove-hydrostatic-code

christopherwharrop-noaa committed:
Merge branch 'develop' into feature/remove-hydrostatic-code

middlecoff committed:
Removed the hydrostatic option which was hard-wired by the declaration
  logical, parameter :: hydrostatic = .false.
This was done to simplify the code.

middlecoff committed:
Merge branch 'feature/remove-hydrostatic-code' of https://github.com/NOAA-GSL/SENA-c_sw into feature/remove-hydrostatic-code

christopherwharrop-noaa reviewed (approved):
Thank you for fixing the indentation.
